kfs-core/src/main/java/edu/arizona/kfs/coa/identity/OrganizationFundReviewRoleTypeServiceImpl.java

```diff
@@ -0,0 +1,119 @@
+                if (roleQualifier.containsKey(KimConstants.AttributeConstants.DOCUMENT_TYPE_NAME)) {
+                    potentialParentDocumentTypeNames.add(roleQualifier.get(KimConstants.AttributeConstants.DOCUMENT_TYPE_NAME));
+                }
+                return potentialParentDocumentTypeNames.isEmpty() 
+                        || StringUtils.equalsIgnoreCase( qualification.get(KimConstants.AttributeConstants.DOCUMENT_TYPE_NAME), roleQualifier.get(KimConstants.AttributeConstants.DOCUMENT_TYPE_NAME)) 
+                        || (KimCommonUtils.getClosestParentDocumentTypeName(getDocumentTypeService().getDocumentTypeByName(qualification.get(KimConstants.AttributeConstants.DOCUMENT_TYPE_NAME)), potentialParentDocumentTypeNames) != null);
+            }             
+        }
+        return false;
+    }
+    
+    /**
+     * comparison for role FG vs. document FG
+     */
+    protected boolean doesFundGroupMatch(Map<String, String> qualification, Map<String, String> roleQualifier) {
+        if (roleQualifier == null || StringUtils.isBlank(roleQualifier.get(KfsKimAttributes.FUND_GROUP_CODE))) {
+            return true;
+        }
+
+        String roleFundGroup = roleQualifier.get(KfsKimAttributes.FUND_GROUP_CODE);        
+
+        if (qualification == null) {
+            return false;
+        }
+        String documentFundGroup = qualification.get(KfsKimAttributes.FUND_GROUP_CODE);
+
+        // check for an exact match
+        return StringUtils.equals(roleFundGroup, documentFundGroup);
+    }
+
+    /**
+     * comparison for role SFG vs. document SFG
+     */
+    protected boolean doesSubFundGroupMatch(Map<String, String> qualification, Map<String, String> roleQualifier) {
+        if (roleQualifier == null || StringUtils.isBlank(roleQualifier.get(KfsKimAttributes.SUB_FUND_GROUP_CODE))) {
+            return true;
+        }
+        String roleSubFund = roleQualifier.get(KfsKimAttributes.SUB_FUND_GROUP_CODE);        
+        
+        if (qualification == null) {
+            return false;
+        }
+        String documentSubFund = qualification.get(KfsKimAttributes.SUB_FUND_GROUP_CODE);
+
+        // check for an exact match
+        if ( StringUtils.equals(roleSubFund, documentSubFund) ) {
+            return true;
+        } else {
+        	// if contains an "*", convert to regex and compare again
+            if (StringUtils.contains(roleSubFund, '*')) {
+                String subFundPattern = StringUtils.replace(roleSubFund, "*", ".*");
+                return documentSubFund.matches(subFundPattern);
+            }
+        }        
+        return false;
+    }
+    
+    public DocumentTypeService getDocumentTypeService() {
+        if (documentTypeService == null) {
+            documentTypeService = SpringContext.getBean(DocumentTypeService.class);
+        }
+        return this.documentTypeService;
+    }
+    
+    public void setDocumentTypeService(DocumentTypeService documentTypeService) {
```

Comment:
This is already removed as I don't see it in my local branch.  I will verify once I commit the other changes that this is removed.